src/PharoLauncher-CLI/PhLImagePackageCliCommand.class.st

```diff
@@ -28,17 +40,120 @@ PhLImagePackageCliCommand class >> launcherCmdPositionals [
+					implicitMeaning: [ :arg :app |NotFound ];
+					yourself).
+		((ClapPositional id: #vmName)
+				description: 'Local vm to put in the package';
+					meaning: [ :pos | pos word asString ];
+					implicitMeaning: [ :arg :app | 'default' ];
+					yourself)
 	}
 ]
 
-{ #category : #creating }
+{ #category : #conversion }
+PhLImagePackageCliCommand >> convertToZip: imagePackageFileReference [ 
+	| zip |
+	zip := ZipArchive new.
+	zip addTree: imagePackageFileReference relativeTo: (imagePackageFileReference parent ) match: [ :e | true ].
+	zip writeToFile:  ( self location, '/' ,self image name,'.zip' ) asFileReference
+]
+
+{ #category : #copying }
+PhLImagePackageCliCommand >> copyFilesOf: anImage at: aDirectory [
+	((anImage file parent childrenMatching: '*.image') at: 1)
+		copyTo: (aDirectory , '/' , anImage name , '.image') asFileReference.
+	((anImage file parent childrenMatching: anImage name,'.sh') at: 1)
+		copyTo: (aDirectory , '/' , anImage name , '.sh') asFileReference.
+	((anImage file parent childrenMatching: anImage name,'.changes') at: 1)
+		copyTo: (aDirectory , '/' , anImage name , '.changes') asFileReference.
+	((anImage file parent parent parent / 'vms' childrenMatching: self vmName) at: 1)
+		copyAllTo: (aDirectory , '/', self vmName) asFileReference
+]
+
+{ #category : #creation }
+PhLImagePackageCliCommand >> createDirectoryAccordingToPath: aPath [ 
+	aPath asFileReference ensureCreateDirectory 
+]
+
+{ #category : #'script generation' }
 PhLImagePackageCliCommand >> createLaunchScript: anImage [
 	anImage launchScriptName
-		writeStreamDo: [ :stream | stream nextPutAll: anImage launchScript ].
-	LibC runCommand: 'chmod +x ',anImage launchScriptName fullName
+		writeStreamDo: [ :stream | stream nextPutAll: (self launchScript: anImage) ].
+	
 ]
 
 { #category : #accessing }
 PhLImagePackageCliCommand >> executeWithImage: anImage [
-	self createLaunchScript: anImage 
+	| userNewDirectory  |
+	self image: anImage.
+	userNewDirectory := self location , '/' , anImage name.
+	self createLaunchScript: anImage.
+	self createDirectoryAccordingToPath: userNewDirectory.
+	[ self copyFilesOf: anImage at: userNewDirectory ]
+		on: FileExists
+		do: [ ^ self outStream
+				nextPutAll: 'a File with the image name already exists at this location';
+				newLine ].
+	LibC
+		runCommand:
+			'find ' , self location , '/"' , anImage name , '"/' , self vmName, ' -type f -exec chmod +x {} \;'.
+	LibC
+		runCommand:
+			'find ' , self location , '/"' , anImage name , '" -iname "', anImage name , '.sh" -type f -exec chmod +x {} \;'.
+			
+	self shouldZip
+		ifTrue: [ self convertToZip: userNewDirectory asFileReference ].
+
+]
+
+{ #category : #accessing }
+PhLImagePackageCliCommand >> image [ 
+	^image 
+]
+
+{ #category : #accessing }
+PhLImagePackageCliCommand >> image: anImage [
+	image := anImage 
+]
+
+{ #category : #'script generation' }
+PhLImagePackageCliCommand >> launchScript: anImage [
+	^ String streamContents: [ :stream |
+		stream
```

Comment:
To allow the script to be run from any folder, you should detect the script folder and add it to the command being run. You should also detect the OS to allow the script to work on OS X.
```bash
# some magic to find out the real location of this script dealing with symlinks
DIR=`readlink "$0"` || DIR="$0";
ROOT=`dirname "$DIR"`;

# DETECT SYSTEM PROPERTIES ======================================================
TMP_OS=`uname | tr "[:upper:]" "[:lower:]"`
if [[ "{$TMP_OS}" = *darwin* ]]; then
    OS="mac";
elif [[ "{$TMP_OS}" = *linux* ]]; then
    OS="linux";
else
    echo "Unsupported OS";
    exit 1;
fi

# RUN THE VM and pass along all arguments as is ================================
if [ "$OS" = "linux" ]; then
    "$ROOT"/pharo-vm/pharo ...
elif [ "$OS" = "mac" ]; then 
    "$ROOT"/Pharo ...
fi